qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -0,0 +1,288 @@
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        routing_pass = BasicSwap(coupling_map)
+        vf2_call_limit = get_vf2_limit(pass_manager_config)
+        if opt_level == 0:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 1:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                check_trivial=True,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 2:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 3:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        else:
+            raise TranspilerError(f"Invalid optimization level specified: {opt_level}")
+
+
+class StochasticSwapPassManager(PassManagerStagePlugin):
+    """Plugin class for routing stage with :class:`~.StochasticSwap`"""
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        vf2_call_limit = get_vf2_limit(pass_manager_config)
+        if opt_level == 0:
+            routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
```

Comment:
If I read this correctly, `StochasticSwap` construction can be moved out of the elif.opt_level 0, 1, and 2 have it. But not 3. Maybe a bad copypaste?

```suggestion
        routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
        if opt_level == 0:
```

Reply:
Fixed as part of: https://github.com/Qiskit/qiskit-terra/pull/8305/commits/58f63065ad5ba16c556048de9bb71ce9f6866b15